src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -304,8 +310,18 @@ def _embed_purpose_or_feature_under_system(
     if embedded_non_vendor_record:
         return
 
-    # Nest new cloned TCF purpose or feature record beneath system otherwise
-    system_section.append(embedded_tcf_record)  # type: ignore[arg-type]
+    if is_purpose_section:
+        # Build the EmbeddedPurpose record with the retention period
+        system_section.append(
+            EmbeddedPurpose(  # type: ignore[arg-type]
+                id=embedded_tcf_record.id,
+                name=embedded_tcf_record.name,
+                retention_period=retention_period,
+            )
+        )
+    else:
+        # Nest new cloned feature record beneath system otherwise
+        system_section.append(embedded_tcf_record)
```

Comment:
Great this is working as I'd expect now, the main thing we fixed is that if the same purpose occurred on multiple vendors, with different retention periods, both of those retention periods show up individually instead of just the first purpose's retention being used.

Can we add a test for this?   Two vendors with the same purpose, but the retention periods differ on both.

Reply:
done! https://github.com/ethyca/fides/pull/4286/commits/38be263dbb0491df54233d1e2941246235a08faa